static/app/components/forms/fields/selectAsyncField.tsx

```diff
@@ -21,94 +21,76 @@ export interface SelectAsyncFieldProps
-    if (!optionObj) {
-      value = optionObj;
-    } else if (this.props.multiple && Array.isArray(optionObj)) {
-      // List of optionObjs
-      value = optionObj.map(({value: val}) => val);
-    } else if (!Array.isArray(optionObj)) {
-      value = optionObj.value;
-    }
-    this.setState({latestSelection: optionObj});
-    onChange?.(value, event);
-    onChangeOption?.(optionObj, event);
-    onBlur?.(value, event);
-  };
-
-  findValue(propsValue: string): GeneralSelectValue {
-    const {defaultOptions} = this.props;
-    const {results, latestSelection} = this.state;
-    // We don't use defaultOptions if it is undefined or a boolean
-    const options = typeof defaultOptions === 'object' ? defaultOptions : [];
-    /**
-     * The propsValue is the `id` of the object (user, team, etc), and
-     * react-select expects a full value object: {value: "id", label: "name"}
-     **/
-    return (
-      // When rendering the selected value, first look at the API results...
-      results.find(({value}) => value === propsValue) ??
-      // Then at the defaultOptions passed in props...
-      options?.find(({value}) => value === propsValue) ??
-      // Then at the latest value selected in the form
-      (latestSelection as GeneralSelectValue)
-    );
-  }
-
-  render() {
-    const {onChangeOption, ...otherProps} = this.props;
-    return (
-      <FormField {...otherProps}>
-        {({
-          required: _required,
-          children: _children,
-          onBlur,
-          onChange,
-          onResults,
-          value,
-          ...props
-        }) => (
+        return (
           <SelectAsyncControl
-            {...props}
-            onChange={this.handleChange.bind(this, onBlur, onChange, onChangeOption)}
+            {...fieldProps}
+            onChange={(option, e) => {
+              const resultValue = !option
+                ? option
+                : props.multiple && Array.isArray(option)
+                ? // List of optionObjs
+                  option.map(({value: val}) => val)
+                : !Array.isArray(option)
+                ? option.value
+                : option;
+
+              setLatestSelection(option);
+              onChange?.(resultValue, e);
+              onChangeOption?.(option, e);
+              onBlur?.(resultValue, e);
+            }}
             onResults={data => {
-              const results = onResults(data);
-              const resultSelection = results.find(result => result.value === value);
-              this.setState(
-                resultSelection ? {results, latestSelection: resultSelection} : {results}
-              );
-              return results;
+              const newResults = onResults(data);
+              const resultSelection = newResults.find(result => result.value === value);
+
+              setResults(newResults);
+              if (resultSelection) {
+                setLatestSelection(resultSelection);
+              }
```

Comment:
`latestSelection` can be `undefined`. would be fine to remove the check and just set it here?